docs/FOR DAPP DEVS/Guides/Query Balances.md

```diff
@@ -0,0 +1,154 @@
+* [JSON-RPC](https://docs.imversed.com/developers/guides/query_balances.html#json-rpc)
+* [gRPC](https://docs.imversed.com/developers/guides/query_balances.html#grpc)
+
+> **Note:** In this document, the command line is used to interact with endpoints. For dApp developers, using libraries such as [cosmjs](https://github.com/cosmos/cosmjs) and [imversedjs](https://docs.imversed.com/developers/libraries/imversedjs.html) is recommended instead.
+
+## `imversed` & Tendermint RPC
+Upon [installation](https://docs.imversed.com/validators/quickstart/installation.html) and [configuration](https://docs.imversed.com/validators/quickstart/binary.html) of the Imversed Daemon, developers can query account balances using `imversed` with the following CLI command:
+
+```text
+$ imversed query bank balances $IMVADDRESS --count-total=$COUNTTOTAL --height=$HEIGHT --output=$OUTPUT --node=$NODE
+balances:
+- amount: "1000000000000000000"
+  denom: aimv
+- amount: "100000"
+  denom: ibc/ED07A3391A112B175915CD8FAF43A2DA8E4790EDE12566649D0C2F97716B8518
+pagination:
+  next_key: null
+  total: "0"
+```
+
+where:
+
+* `$IMVADDRESS` is the Imversed address with balances of interest (eg. `imversed1...`).
+* (optional) `$COUNTTOTAL` counts the total number of records in all balances to query for.
+* (optional) `$HEIGHT` is the specific height to query state at (can error if node is pruning state).
+* (optional) `$OUTPUT` is the output format (eg. `text`).
+* (optional if running local node) `$NODE` is the Tendermint RPC node information is requested from (eg. `https://tendermint.bd.imversed.org:26657`).
+
+Details of non-native currencies (ie. not `aimv`) can be queried with the following CLI command:
+
+```text
+$ imversed query erc20 token-pair $DENOM --node=$NODE --height=$HEIGHT --output=$OUTPUT
+token_pair:
+  contract_owner: OWNER_MODULE
+  denom: ibc/ED07A3391A112B175915CD8FAF43A2DA8E4790EDE12566649D0C2F97716B8518
+  enabled: true
+  erc20_address: 0xFA3C22C069B9556A4B2f7EcE1Ee3B467909f4864
+```
+
+where `$DENOM` is the denomination of the coin (eg. `ibc/ED07A3391A1...`).
+
+## JSON-RPC
+
+Developers can query account balances of `aimv` using the [`eth_getBalance`](https://docs.imversed.com/developers/json-rpc/endpoints.html#ethgetbalance) JSON-RPC method in conjunction with [`curl`](https://curl.se/):
+
+where:
+
+* `$ETHADDRESS` is the Etherum hex-address the balance is to be queried from. Note that Imversed addresses (those beginning with `imversed1...`) can be converte.d to Ethereum addresses using libraries such as [imversedjs](https://docs.imversed.com/developers/libraries/imversedjs.html).
+* `$BLOCK` is the block number or block hash (eg. `"0x0"`). The reasoning for this parameter is due to [EIP-1898](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1898.md).
+* (optional if running local node) `$NODE` is the JSON-RPC node information is requested from (eg. `https://eth.bd.imversed.org:8545`).

```

Comment:
We don't have that address: https://eth.bd.imversed.org:8545

You can contact us in discord to get seed.